javalanglib/src/main/scala/java/lang/Long.scala

```diff
@@ -114,6 +114,48 @@ object Long {
     }
   }
 
+  // Ported from https://github.com/gwtproject/gwt/blob/master/user/super/com/google/gwt/emul/java/lang/Long.java
+  def toString(value: scala.Long, intRadix: Int): String = {
+    var _value: scala.Long = value
+
+    if (intRadix == 10 || intRadix < Character.MIN_RADIX || intRadix > Character.MAX_RADIX) {
+      toString(value)
+    }
+    else if (Integer.MIN_VALUE <= _value && (_value <= Integer.MAX_VALUE)) {
+      Integer.toString(_value.toInt, intRadix)
+    }
+    else {
+      def forDigit(digit: Int): Char = {
+          val overBaseTen = digit - 10
+          (if (overBaseTen < 0) '0' + digit else 'a' + overBaseTen).toChar
+      }
+
+      val bufSize: Int = 65
+      val buf = Array.ofDim[Char](bufSize)
+      var pos: Int = bufSize - 1
+      var radix: scala.Long = intRadix
+      if (_value >= 0) {
+        while (_value >= radix) {
+          buf(pos) = forDigit((_value % radix).toInt)
+          pos -= 1
+          _value /= radix
+        }
+        buf(pos) = forDigit(_value.toInt)
+      } else {
+        val negRadix: scala.Long = -radix
+        while (_value <= negRadix) {
+          buf(pos) = forDigit(-(_value % radix).toInt)
+          pos -= 1
+          _value /= radix
+        }
+        buf(pos) = forDigit(-_value.toInt)
+        pos -= 1
+        buf(pos) = '-'
```

Comment:
I'm not so sure on this one. Looking at the non-radix version, it makes sense how that was done. Here it is not so clean as both the radix and the value are negated, so again I'm not so convinced on the benefit. I would be tempted to keep as is (mainly as it is a very low level class) - your call.